File: app/src/main/java/tigerhacks/android/tigerhacksapp/sponsors/Sponsor.java
package tigerhacks.android.tigerhacksapp.sponsors;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Sponsor {

    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("mentors")
    @Expose
    private List<Mentor> mentors = null;
    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("website")
    @Expose
    private String website;
    @SerializedName("location")
    @Expose
    private String location;
    @SerializedName("image")
    @Expose
    private String image;
    @SerializedName("level")
    @Expose
    private String level;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Mentor> getMentors() {
        return mentors;
    }

    public void setMentors(List<Mentor> mentors) {
        this.mentors = mentors;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getWebsite() {
        return website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

}

File: app/src/main/kotlin/tigerhacks/android/tigerhacksapp/sponsors/Sponsor.kt
package tigerhacks.android.tigerhacksapp.sponsors

import android.os.Parcelable
import androidx.annotation.ColorRes
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.squareup.moshi.JsonClass
import kotlinx.android.parcel.Parcelize
import tigerhacks.android.tigerhacksapp.R

/**
 * @author pauldg7@gmail.com (Paul Gillis)
 */
@Entity
@JsonClass(generateAdapter = true)
@Parcelize
data class Sponsor(
    @PrimaryKey val name: String = "",
    val description: String = "",
    val website: String = "",
    val location: String = "",
    val image: String = "",
    val level: String = ""
) : Parcelable {
    @ColorRes
    fun getSponsorLevelColorRes(): Int {
        return when (level) {
            "Platinum" -> R.color.platinum
            "Gold" -> R.color.gold
            "Silver" -> R.color.silver
            "Bronze" -> R.color.bronze
            else -> R.color.colorPrimary
        }
    }

    fun getLevelNum() = when (level) {
        "Platinum" -> 0
        "Gold" -> 1
        "Silver" -> 2
        else -> 3
    }
}